x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/rule_management/crud/create_rule/request_schema_validation.ts

```diff
@@ -5,6 +5,7 @@
  * 2.0.
  */
```

Comment:
I wonder if we can utilize Open API schema for fields length validation in `x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/model/rule_schema/specific_attributes/threshold_attributes.schema.yaml`

We already use it for limiting alert suppression fields length
https://github.com/elastic/kibana/blob/9.1/x-pack/solutions/security/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.schema.yaml#L735-L740

That would help to eliminate some manual checks on each API endpoint.

Reply:
Nice! I have updated the openapi spec and removed this validation from the code since it is already present in the generated schema. I tested this by calling the POST, PUT and PATCH endpoints on the rule.